test/deprecates.jl

```diff
@@ -0,0 +1,37 @@
+@testset "Deprecations" begin
+    @testset "Syslog" begin
+        if Sys.isunix()
+            levels = copy(Memento._log_levels)
+            levels["invalid"] = 100
+
+            # Create our DefaultHandler w/ the Syslog IO type
+            handler = DefaultHandler(Memento.Syslog(), DefaultFormatter("{level}: {msg}"))
```

Comment:
FYI:
```julia
julia> Array{Int}(1)
WARNING: Array{T}(m::Int) where T is deprecated, use Array{T}(uninitialized, m) instead.
Stacktrace:
 [1] depwarn at ./deprecated.jl:68 [inlined]
 [2] Array{Int64,N} where N(::Int64) at ./deprecated.jl:56
 [3] top-level scope
 [4] eval(::Module, ::Expr) at ./repl/REPL.jl:3
 [5] eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./repl/REPL.jl:69
 [6] macro expansion at ./repl/REPL.jl:100 [inlined]
 [7] (::getfield(Base.REPL, Symbol("##1#2")){Base.REPL.REPLBackend})() at ./event.jl:95
in expression starting at no file:0
1-element Array{Int64,1}:
 4596391456
```

```julia
julia> using Test

julia> @test_warn "Array{T}(m::Int) where T is deprecated, use Array{T}(uninitialized, m) instead." Array{Int}(1)
1-element Array{Int64,1}:
 4520842800
```
Unfortunately I think this is only on Julia 0.7

Reply:
I should be able to use `@test_warn` with `Compat.Test`.